api/python/provisioner/commands/cluster_id.py

```diff
@@ -21,22 +21,77 @@
 import uuid
 from typing import Type
 
+from . import (
+     CommandParserFillerMixin
+)
 from .. import inputs
 from ..api import grains_get
-from ..config import LOCAL_MINION
+from ..config import (
+     LOCAL_MINION,
+     ALL_MINIONS
+)
+from ..vendor import attr
 
 from provisioner.commands import (
-     PillarSet,
-     GetClusterId
+     PillarGet,
+     PillarSet
 )
-from . import CommandParserFillerMixin
-
-from ..vendor import attr
 from provisioner.salt import local_minion_id
 
 logger = logging.getLogger(__name__)
 
 
+@attr.s(auto_attribs=True)
+class GetClusterId(CommandParserFillerMixin):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+
+    @staticmethod
+    def run(targets=ALL_MINIONS):
+        """get_cluster_id command execution method.
+
+        Gets cluster ID from pillar data.
+
+        Execution:
+        `provisioner get_cluster_id`
+        Takes no mandatory argument as input.
+
+        """
+        try:
+            cluster_data = PillarGet().run(
+                'cluster',
+                targets=targets
+            )
+            cluster_id = []
+
+            for node in cluster_data:
+                if "cluster_id" in cluster_data[node]["cluster"]:
+                    cluster_id.append(
+                      cluster_data[node]["cluster"]["cluster_id"]
+                    )
+                else:
+                    raise ValueError(
+                         "ClusterID is not present in Pillar data for "
+                         "either of the nodes. To set one, execute "
+                         "`provisioner set_cluster_id` command."
+                    )
+
+            if len(set(cluster_id)) != 1:
+                raise ValueError(
+                     "Error: ClusterID assignment not unique across "
+                     f"the nodes of the cluster: {cluster_id}. "
+                     "To set one, execute `provisioner set_cluster_id` command."
+                )
+
+            return cluster_id[0]
+
+        except ValueError as exc:
+            logger.error(
```

Comment:
I don't think we need to hide all exceptions here. I would suggest to use `raise` instead of `return None` here

Reply:
@andkononykhin The reason I moved `raise` to log error and `return None` is because we're calling this api in `set_cluster_id` , which in turn, is called at the end of bootstrap process (`setup_provisioner`). So, when it returns `None` as against an exception, we set the value to pillar as part of `SetClusterId` class. 

Can we handle this flow any other way.?